nucypher/policy/conditions/offchain.py

```diff
@@ -0,0 +1,142 @@
+from typing import Any, Optional, Tuple
+
+import requests
+from jsonpath_ng.exceptions import JsonPathLexerError, JsonPathParserError
+from jsonpath_ng.ext import parse
+from marshmallow import ValidationError, fields, post_load, validate
+from marshmallow.fields import Field
+
+from nucypher.policy.conditions.base import AccessControlCondition
+from nucypher.policy.conditions.exceptions import (
+    ConditionEvaluationFailed,
+    InvalidCondition,
+)
+from nucypher.policy.conditions.lingo import ConditionType, ReturnValueTest
+from nucypher.policy.conditions.utils import CamelCaseSchema
+from nucypher.utilities.logging import Logger
+
+
+class JSONPathField(Field):
+    default_error_messages = {"invalid": "Not a valid JSONPath expression."}
+
+    def _deserialize(self, value, attr, data, **kwargs):
+        if not isinstance(value, str):
+            self.fail("invalid")
+        try:
+            parse(value)
+        except (JsonPathLexerError, JsonPathParserError):
+            self.fail("invalid")
+        return value
+
+
+class OffchainCondition(AccessControlCondition):
+    """
+    An offchain condition is a condition that can be evaluated by reading from a JSON
+    endpoint. This may be a REST service but the only requirement is that
+    the response is JSON and can be parsed using jsonpath.
+    """
+
+    CONDITION_TYPE = ConditionType.OFFCHAIN.value
+    LOGGER = Logger("nucypher.policy.conditions.offchain")
+
+    class Schema(CamelCaseSchema):
+
+        name = fields.Str(required=False)
+        condition_type = fields.Str(
+            validate=validate.Equal(ConditionType.OFFCHAIN.value), required=True
+        )
+        headers = fields.Dict(required=False)
+        parameters = fields.Dict(required=False)
+        endpoint = fields.Str(required=True)
```

Comment:
What do you think about adding a facility (in this PR or just in general) to restrict access to certain offchain elements. I think less permissive is better (an allowlist). One example of restricted-by-default element could be an `http://` request. We could restrict all headers by default too. 

This is something we should discuss in respect to the security of offchain evaluation.

Reply:
> Perhaps we just don't expose headers at all?

Maybe not until it's functionally necessary?

> With regard to HTTP requests - let's just block them altogether?

I think so

Reply:
✔️